mautic/mautic
# Allow admin to disable background ajax calls on global search until user hits enter

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | No
| New feature? | Yes
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

On systems with large amounts of data, having the global search kickoff background search suggestions on 3 typed chars can take a while. This allows an admin to toggle a setting on the System configuration page to disable the background ajax calls until the user hits enter.

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. 
2. 

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Set the "Disable Search Until Enter" flag to True in system configurations.
3. from any page that contains the global search widget, type 1-3 characters, including a space, and verify no search ajax request occurs.
4. Type 4 or more characters and verify no ajax search request occurs.
Hit Enter, verify search ajax request occurs and a suggestion modal pops up (standard behavior).

#### List deprecations along with the new alternative:
1. 
2. 

#### List backwards compatibility breaks:
1. 
2.
